### DISCUSSION TITLE
Extend Heading card functionality 1. Add support for templates, 2. Add support for multi-line Title and Subtitle

### DISCUSSION THREAD

**Discussion by circa1665** [home-assistant/frontend#22243] [Dashboard]
### 1. Add support for templates
Would it be possible for the Heading card to support templates? Similar to the [Mushroom title card](https://github.com/piitaya/lovelace-mushroom/blob/main/docs/cards/title.md)?
Would like to see the following support templates:
1. Icon
2. Title
3. Subtitle

This could just be in yaml for advanced users, or if the fields in the popup modal could be made slightly larger so there's room to add/edit templates in the visual editor?
<img width="1386" alt="Screenshot 2024-10-04 at 22 52 26" src="https://github.com/user-attachments/assets/7aa3d22b-4182-4316-ab4b-6bf9a2101eae">

### 2. Add support for multi-line Title and Subtitle
Would it be possible for the Heading card to support multi-line Title and Subtitle?

**Comment by BBE-FR** [comment]
Fully support this proposal !

As a temporary workaround, it is possible to use a dedicated helper for this, but only at the end of the section, not in the name space, nor in the icon part.
![image](https://github.com/user-attachments/assets/f3302247-5545-4c45-8e76-c8ca67922617)

**Reply by BBE-FR** [reply]
Also it is possible to customise a little bit the icons for the entities displayed with themes using state--color
https://www.home-assistant.io/integrations/frontend#state-color
```
  #door
  state-binary_sensor-door-off-color: green
  state-binary_sensor-door-on-color: red
  #motion
  state-binary_sensor-motion-off-color: green
  state-binary_sensor-motion-on-color: red
```

But this does not allow templating options (or you have to put the template into a dedicated helper)
And this does not affect the "main" icon, only the entities one (on the right)
![image](https://github.com/user-attachments/assets/c672b616-1726-4b54-bddc-9d0e2f75fcbe)
